crates/bin/pindexer/src/dex_ex/mod.rs

```diff
@@ -982,6 +990,107 @@ impl Component {
         Ok(())
     }
 
+    async fn record_swap_execution_traces(
+        &self,
+        dbtx: &mut PgTransaction<'_>,
+        time: DateTime,
+        height: i32,
+        swap_execution: &SwapExecution,
+    ) -> anyhow::Result<()> {
+        let SwapExecution {
+            traces,
+            input: se_input,
+            output: se_output,
+        } = swap_execution;
+
+        let asset_start = se_input.asset_id;
+        let asset_end = se_output.asset_id;
+        let batch_input = se_input.amount;
+        let batch_output = se_output.amount;
+
+        for trace in traces.iter() {
+            let Some(input_value) = trace.first() else {
+                continue;
+            };
+            let Some(output_value) = trace.last() else {
+                continue;
+            };
+
+            let input = input_value.amount;
+            let output = output_value.amount;
+
+            let price_float = (output.value() as f64) / (input.value() as f64);
+            let amount_hops = trace
+                .iter()
+                .map(|x| BigDecimal::from(x.amount.value()))
+                .collect::<Vec<_>>();
+            let position_id_hops: Vec<[u8; 32]> = vec![];
+            let asset_hops = trace
+                .iter()
+                .map(|x| x.asset_id.to_bytes())
+                .collect::<Vec<_>>();
+
+            sqlx::query(
+                "INSERT INTO dex_ex_batch_swap_traces (
+                height,
+                time,
+                input,
+                output,
+                batch_input,
+                batch_output,
+                price_float,
+                asset_start,
+                asset_end,
+                asset_hops,
+                amount_hops,
+               position_id_hops, 
```

Comment:
```suggestion
               position_id_hops
```

I think this should fix the SQL error (no trailing commas allowed :()